### ISSUE TITLE
Possible false positive cyclical import

### ISSUE THREAD

**Issue by Gjgarr** [microsoft/pyright#3278]
**Describe the bug**
Pyright reports a cyclic import within `__init__.py` when a star-import includes an imported module at the same level.

**To Reproduce**
Reported as cyclical import:
```python
# ./cyclical/__init__.py
from .f1 import * # Cycle detected - __init__.py & f1.py

# ./cyclical/f1.py
from . import f2

# ./cyclical/f2.py (empty)
```

No cyclical import reported:
```python
# ./not_cyc/__init__.py (empty)

# ./not_cyc/f1.py
from .f2 import *

# ./not_cyc/f2.py
from . import f3

# ./not_cyc/f3.py (empty)
```

**Expected behavior**
Neither scenario reported as cyclical.

**VS Code extension or command-line**
VS Code extension - pylance v2022.3.4

**Comment by erictraut** [comment]
This is intended behavior, not a false positive.

The statement `from .f1 import *` within an `__init__.py` method imports `f1` and assigns a reference to the `f1` module to the local symbol `f1`. The `f1` module imports `f2` from the `__init__.py` module. So this is an actual cyclical import.

The cycle detection code in pyright is admittedly strict about what constitutes an import cycle. It will detect cycles that do not necessarily result in runtime exceptions. That's intended. If you want to use this diagnostic check, you'll need to factor your code to avoid all cycles.
